core/vdbe/mod.rs

```diff
@@ -260,6 +261,13 @@ pub enum Insn {
         dest: usize,      // P3
         func: ScalarFunc, // P4
     },
+
+    // Json
+    Json {
+        start_reg: usize,
+        dest: usize,
+        func: JsonFunc,
+    },
```

Comment:
The JSON function calls should use the `Function` instruction like other functions do:

```
sqlite> EXPLAIN SELECT json('{}');
addr  opcode         p1    p2    p3    p4             p5  comment
----  -------------  ----  ----  ----  -------------  --  -------------
0     Init           0     7     0                    0   Start at 7
1     Once           0     4     0                    0
2     String8        0     3     0     {}             0   r[3]='{}'
3     Function       1     3     2     json(1)        0   r[2]=func(r[3])
4     Copy           2     1     0                    0   r[1]=r[2]
5     ResultRow      1     1     0                    0   output=r[1]
6     Halt           0     0     0                    0
7     Goto           0     1     0                    0
```

Reply:
Thanks for your feedback. I've removed `Insn::JsonFunc` and updated `Insn::Function` to receive a new enum. I added JsonFunc initially because there are many JSON-related functions in SQLite3. What do you think?

Reply:
Thanks perfect, @JeanArhancet! Can you add one or two TCL tests to ensure the function behaves the same as with SQLite and let's get this merged!

Reply:
@penberg I have added some TCL tests based on the SQLite tests. If I am not mistaken, I have noticed queries not supported by the crate we are using, such as `Infinity` (https://github.com/sqlite/sqlite/blob/master/test/json501.test#L254), which returns null (which is why I did not include it in the TCL tests). Another case I have noticed is hexadecimal values with a +/- sign that are not supported (https://github.com/sqlite/sqlite/blob/master/test/json501.test#L194).